app/src/main/java/com/eveningoutpost/dexdrip/LibreReceiver.java

```diff
@@ -69,7 +69,8 @@ public void run() {
                                 Libre2RawValue currentRawValue = processIntent(intent);
                                 if (currentRawValue == null) return;
                                 Log.v(TAG,"got bg reading: from sensor:"+currentRawValue.serial+" rawValue:"+currentRawValue.glucose+" at:"+currentRawValue.timestamp);
-                                if(!BgReading.last_within_minutes(5)) {
+                                // period of 4.5 minutes to collect 5 readings
```

Comment:
Not sure, but if you decrease the time frame of collecting the bg-Data to less then 5 minutes bevoe smoothing and storing, why you expect that you will get safely 5 readings? As far as I have seen the intends are every minute (between 59 up to 62 seconds), so in my mind, if you decrease the time frame to lower then 5, you will maximum work only with the last 4 readings in addition to the 20 minutes of smoothing?

Reply:
It was proofed via extensive testing. With the 5 minutes frame the valuecount per interval was changing between 4 and 6. Going to 4.5 I got mostly 5 readings.

Reply:
Ok, so then it should be tested in the wild ( I also did not think that it will make a big difference to have 4 or 5 readings ;) ) we will see.

Reply:
Having a new value every 5 minutes instead of jumping between 5 and 6 minutes intervals makes a difference in looping. So making sure it is recorded exactly every 5 minutes by setting the time to not 5 minutes but slightly less has a real advantage here!

Reply:
@MasterPlexus: if you wait more than 4, you will get the 5th reading in. If you wait 5 minutes, you may still reject the 5th reading and not create a GlucoseValue.

Reply:
OK, so the comment is also slight misunderstoodable, as it suggests to get 5 readings, but means to be save to have a calculated value every 5 minutes... So sorry for the notice, was just wondering, but I'm clearly fine with it.